.github/scripts/wasm-target-test-build.sh

```diff
@@ -0,0 +1,17 @@
+GIT_ROOT=$(pwd)
+
+cd /tmp
+
+rustup default stable
+rustup target add wasm32-unknown-unknown wasm32-wasi
+
+cargo new foobar
+cd foobar
+cargo add --path "${GIT_ROOT}/halo2_proofs" --features batch,dev-graph,gadget-traces
+cargo add getrandom --features js
```

Comment:
Technically this feature is only required for `wasm32-unknown-unknown`; AFAIK `wasi` already gives an API to access randomness.  Can you try with this instead?
```suggestion
cargo add getrandom --features js --target wasm32-unknown-unknown
```